lib/Document.php

```diff
@@ -191,14 +195,57 @@ public function download(?bool $postProcess = true, ?string $encoding = null): s
         return $contents;
     }
 
+    /**
+     * Downloads the document data as a stream.
+     * 
+     * @param resource|string|StreamInterface|null $output Optionnaly copy data stream to the given output.
+     *
+     * @param ?string &$encoding Get the document encoding from the HTTP request.
+     *
+     * @return StreamInterface The raw (unencrypted) document stream..
+     */
+    public function downloadStream($output = null, string &$encoding = null): StreamInterface {
+        try {
+            $response = $this->client->request('GET', $this->url, ['stream' => true]);
+        } catch (\GuzzleHttp\Exception\ClientException $e) {
+            $response = $e->getResponse();
+            if ($response->getStatusCode() == 404) {
+                throw new RuntimeException("Document Report not Found ({$response->getStatusCode()}): {$response->getBody()}");
+            } else {
+                throw $e;
+            }
```

Comment:
this `else` statement is redundant, since the `if` case results in an exception being thrown (so the remainder of the containing `catch` block will never be reached. can be simplified to:

```php
if ($response->getStatusCode() === 404) {
    // ...
}
throw $e;
```